### PR TITLE
Unskipped tests in `test_object_spilling`, `test_object_spilling_2`, `test_get_locations`

### PR DESCRIPTION
**Pull request by czgdp1807**
## Why are these changes needed?

Mostly cluster tests are enabled in this PR in the above mentioned files. Some non-cluster tests are also enabled. All of these pass on my machine without issues.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_object_spilling.py [hunk 1]
```diff
@@ -265,10 +261,6 @@ def test_spill_objects_automatically(object_spilling_config, shutdown_only):
     assert_no_thrashing(address["address"])
 
 
-@pytest.mark.skipif(
-    platform.system() in ["Darwin", "Windows"],
-    reason="Failing on Windows, very flaky on OSX.",
-)
```
Comment (czgdp1807) [inline python/ray/tests/test_object_spilling.py:271]: I will skip these on `Darwin` after the current test suite completes.

File: python/ray/cluster_utils.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 logger = logging.getLogger(__name__)
 
-cluster_not_supported = os.name == "nt"
+cluster_not_supported = False
```
Comment (czgdp1807) [inline python/ray/cluster_utils.py:19]: Will remove this change as well. Instead I will ignore `cluster_not_supported` in the pytest fixture, `ray_cluster_start`.

**Comment by pcmoritz** [comment]
There seem to be a lot of failures on the windows CI for this PR, can you look at them?

**Comment by czgdp1807** [comment]
Yes. I will look at these tomorrow.

**Comment by czgdp1807** [comment]
All the failures are `TIMEOUT`s and that too tests which aren't unskipped here. Though its interesting that so many tests timed out in one CI run. I will check locally.

File: python/ray/tests/test_object_spilling.py [hunk 3]
```diff
@@ -158,7 +156,6 @@ def task():
     ray.get([task.remote() for _ in range(2)])
 
 
-@pytest.mark.skipif(platform.system() == "Windows", reason="Failing on Windows.")
 def test_spilling_not_done_for_pinned_object(object_spilling_config, shutdown_only):
```
Comment (czgdp1807) [inline python/ray/tests/test_object_spilling.py:159]: This debug mode of this test hangs in bazel locally and on CI.

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**